modules/data.land/R/cohort2pool.R

```diff
@@ -0,0 +1,55 @@
+##' @name cohort2pool
+##' @title cohort2pool
+##' @export
+##' @param veg_file standard cohort veg_file
+##' @param allom_param parameters for allometric equation, a and b
+##' @author Saloni Shah
+
+cohort2pool <- function(veg_file,allom_param =NULL) {
+  
+  dat <- readRDS(veg_file)
+  
+  #Building Site ID from past directories
+  
+  path <- dirname(veg_file)
+  last_dir <- basename(path)
+  nums_id <- strsplit(last_dir,"[^[:digit:]]")
+  base_id <- nums_id[[1]][length(nums_id[[1]])]
+  suffix <- nums_id[[1]][(length(nums_id[[1]])-1)]
+  if (suffix == 1) {
+    siteid <- as.numeric(paste0(suffix, "0000", base_id))
+  } else if (siteid ==0) {
+    siteid <- as.numeric(base_id)
+  }
+  
+  #Grab DBH
+  
+  dbh <- dat[[2]]$DBH
+  if(is.null(allom_param)){
+    a = 2                        
+    b= 0.3
+  }
+  
+  #Calculate AGB
+  
+  for (i in 1: length(dbh)) {
+    biomass = a + b*log(dbh)  
```

Comment:
1. use seq_along(dbh) not 1:length
2. most allometries are base 10
3. if you're calculating the allometry in the log domain, you need to unlog the result (i.e. ```biomass = 10^(a+b*log10(dbh))``` though do check that the ```a``` you are using is appropriate, and not actually the linear domain parameter ```biomass = a*dbh^b``` -- the default you use sure looks like this version. 
4. you need to calculate more than just AGB!  Where's foliar biomass for LAI? Roots?